manifests/router/service.pp

```diff
@@ -3,15 +3,23 @@
 # @api private
 class qpid::router::service {
 
+  if $qpid::router::ensure == 'absent' {
+    $_service_ensure = false
+    $_service_enable = false
+  } else {
+    $_service_ensure = $qpid::router::service_ensure
+    $_service_enable = $qpid::router::service_enable
+  }
+
   service { 'qdrouterd':
-    ensure     => $qpid::router::service_ensure,
-    enable     => pick($qpid::router::service_enable, $qpid::router::service_ensure),
+    ensure     => $_service_ensure,
+    enable     => pick($_service_enable, $_service_ensure),
     hasstatus  => true,
     hasrestart => true,
   }
 
   if $facts['systemd'] {
-    if $qpid::router::open_file_limit {
+    if $qpid::router::open_file_limit and $qpid::router::ensure {
```

Comment:
Any string in Puppet evaluates to true so it should have also checked for an equality to `present`.